pymmcore_plus/core/_mmcore_plus.py

```diff
@@ -447,81 +447,21 @@ def getCameraChannelNames(self) -> Tuple[str, ...]:
 
-    def _mda(self, sequence) -> None:
-        self.events.sequenceStarted.emit(sequence)
-        logger.info("MDA Started: {}", sequence)
-        self._paused = False
-        paused_time = 0.0
-        t0 = time.perf_counter()  # reference time, in seconds
-
-        def check_canceled():
-            if self._canceled:
-                logger.warning("MDA Canceled: {}", sequence)
-                self.events.sequenceCanceled.emit(sequence)
-                self._canceled = False
-                return True
-            return False
-
-        for event in sequence:
-            while self._paused and not self._canceled:
-                paused_time += 0.1  # fixme: be more precise
-                time.sleep(0.1)
-
-            if check_canceled():
-                break
-
-            if event.min_start_time:
-                go_at = event.min_start_time + paused_time
-                # We need to enter a loop here checking paused and canceled.
-                # otherwise you'll potentially wait a long time to cancel
-                to_go = go_at - (time.perf_counter() - t0)
-                while to_go > 0:
-                    while self._paused and not self._canceled:
-                        paused_time += 0.1  # fixme: be more precise
-                        to_go += 0.1
-                        time.sleep(0.1)
-
-                    if self._canceled:
-                        break
-                    if to_go > 0.5:
-                        time.sleep(0.5)
-                    else:
-                        time.sleep(to_go)
-                    to_go = go_at - (time.perf_counter() - t0)
-
-            # check canceled again in case it was canceled
-            # during the waiting loop
-            if check_canceled():
-                break
-
-            logger.info(event)
-
-            # prep hardware
-            if event.x_pos is not None or event.y_pos is not None:
-                x = event.x_pos or self.getXPosition()
-                y = event.y_pos or self.getYPosition()
-                self.setXYPosition(x, y)
-            if event.z_pos is not None:
-                self.setZPosition(event.z_pos)
-            if event.channel is not None:
-                self.setConfig(event.channel.group, event.channel.config)
-            if event.exposure is not None:
-                self.setExposure(event.exposure)
-
-            # acquire
-            self.waitForSystem()
-            self.snapImage()
-            img = self.getImage()
-
-            self.events.frameReady.emit(img, event)
-
-        logger.info("MDA Finished: {}", sequence)
-        self.events.sequenceFinished.emit(sequence)
+    def register_mda_engine(self, engine):
+        if not isinstance(engine, PMDAEngine):
+            raise TypeError("engine must conform to the Engine protocol.")
+        previous_engine = self._mda_engine
+        self._mda_engine = engine
+        self.events.mdaEngineRegistered.emit(engine, previous_engine)

```

Comment:
added so that downstream things like napari-micromanager can unregister their callbacks with the old engine and attach to the new one.